source/Halibut.TestProxy/ProxyConnectionService.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Halibut.TestProxy
+{
+    interface IProxyConnectionService
+    {
+        IProxyConnection GetProxyConnection(ProxyEndpoint endpoint);
+    }
+
+    class ProxyConnectionService : IProxyConnectionService, IHostedService
+    {
+        readonly IProxyConnectionFactory proxyConnectionFactory;
+        readonly ILogger<ProxyConnectionService> logger;
+        readonly ConcurrentDictionary<ProxyEndpoint, IProxyConnection> proxies = new();
```

Comment:
What is the purpose of being keyed by `ProxyEndpoint` does it prevent to clients using this that want to proxy to the same endpoint?